src/sonic-config-engine/sonic-cfggen

```diff
@@ -229,6 +243,8 @@ def main():
     group.add_argument("--preset", help="generate sample configuration from a preset template", choices=get_available_config())
     group = parser.add_mutually_exclusive_group()
     group.add_argument("-K", "--key", help="Lookup for a specific key")
+    group.add_argument("-b", "--batch", help="Render and dump template data into a file (batch mode)", action="append", default=[],
```

Comment:
>-b [](start = 24, length = 2)

Could you achieve this by bash redirection without any code change? show me some example if you disagree.

Reply:
Here an example:

```
./sonic-cfggen-new -d -j ztp_input.json -b /usr/share/sonic/templates/90-dhcp6-systcl.conf.j2,/home/admin/90-dhcp6-systcl.conf.new -b /usr/share/sonic/templates/dhclient.conf.j2,/home/admin/dhclient.conf.new
```

Reply:
The -b param is a list and could be as many as we want. the -t is just one time call as it dumps the output to the stdout and so in case of we have multiple template to be written to different files we need to call multiple times.

Another solution is we could have a template that include all other template, however breaking the out into corresponding files is a challenge.

Reply:
Thanks for the sample.
With this use case, I propose a command line:
```
./sonic-cfggen-new -d -j ztp_input.json -t /usr/share/sonic/templates/90-dhcp6-systcl.conf.j2,/usr/share/sonic/templates/dhclient.conf.j2 -o /home/admin/90-dhcp6-systcl.conf.new,/home/admin/dhclient.conf.new
```

The benefit is consistent usage for single pair and batch mode. The implementation gaps:
1. -o
2. support multiple input files in -t
3. need to check the file number in -o is same as in -t

---
In reply to: [453047597](https://github.com/Azure/sonic-buildimage/pull/4941#discussion_r453047597) [](ancestors = 453047597)